src/Apollo/NcrReactionsProjector.php

```
@@ -0,0 +1,216 @@
+            ],
+            'UpdateExpression'          =>  rtrim('set'. $updateExpression, ', '),
+            'ExpressionAttributeNames'  => $expressionAttributeNames,
+            'ExpressionAttributeValues' => [
+                ':v_incr' => ['N' => '1'],
+            ],
+            'ReturnValues'              => 'NONE',
+        ];
+
+        $this->client->updateItem($params);
+        $this->ncrSearch->indexNodes([$this->ncr->getNode($reactionsRef, true, $context)], $context);
+    }
+
+    protected function projectNode(Message $reactions, Message $event, Pbjx $pbjx): void
+    {
+        $context = ['causator' => $event];
+        $reactions
+            ->set('updated_at', $event->get('occurred_at'))
+            ->set('updater_ref', $event->get('ctx_user_ref'))
+            ->set('last_event_ref', $event->generateMessageRef())
+            ->set('etag', Aggregate::generateEtag($reactions));
+
+        $this->ncr->putNode($reactions, null, $context);
+        $this->ncrSearch->indexNodes([$reactions], $context);
+        $pbjx->trigger($reactions, 'projected', new NodeProjectedEvent($reactions, $event), false, false);
+    }
+
+    protected function createReactions(NodeRef $nodeRef): Message
+    {
+        static $class = null;
+        if (null === $class) {
+            $class = MessageResolver::resolveCurie('*:apollo:node:reactions:v1');
+        }
+
+        $reactions = $class::fromArray(['_id' => $nodeRef->getId()]);
+        $this->addReactionTypes($reactions);
+
+        return $reactions;
+    }
+
+    protected function getReactions(NodeRef $nodeRef, Message $event): ?Message
+    {
+        try {
+            $reactionsRef = $this->createReactionsRef($nodeRef);
+            return $this->ncr->getNode($reactionsRef, true, ['causator' => $event]);
+        } catch (NodeNotFound $nf) {
+            // Node not found do nothing here
+        } catch (\Throwable $e) {
+            throw $e;
```

@bhsiao55 why catch and immediately throw? seems like you could just catch NodeNotFound and return null for that, anything else let it throw and bubble up